kani-compiler/src/kani_middle/transform/check_aliasing/function_cache.rs

```diff
@@ -0,0 +1,66 @@
+pub use stable_mir::ty::GenericArgKind as GenericArg;
+pub use stable_mir::ty::GenericArgs;
+use super::{MirInstance, MirError, TyCtxt, super::super::find_fn_def};
+/// FunctionSignature encapsulates the data
+/// for rust functions with generic arguments
+/// to ensure that it can be cached.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Signature {
```

Comment:
Is there any reason why you wouldn't use `FnDef`?

Reply:
I could use FnDef instead in the signature, however, I'm trying to cache different generic versions of the same function.

Reply:
I believe that FnDef doesn't encapsulate the generic arguments.

Reply:
Perhaps a combination of FnDef + GenericArgs would work?

Reply:
Alternatively, you could rename `name` to `diagnostic` to make it clear that this is the name of the diagnostic item